### PR TITLE
Upstream merge of the fork^2 of supabase-py

### PR DESCRIPTION
**Pull request by fedden**
Hey so a bunch of small changes, to ensure we are working off the same source code. Mostly cosmetic changes, sorry about that, but was just working through each line of code and acquainting myself with it. 

- Adds the create_client method like in the supabase-js library
- Adds some tests that currently fail, will add more in due course. 
- Changes how we interface with the auth client - prefers kwargs over dicts of parameters as more pythonic
- Changes variable and function names to be snake_case
- Adds a little more to the docstring
- Enables `pip install -e .` installs with a setup.py stub

### CODE REVIEW

File: supabase_py/client.py [hunk 1]
```diff
@@ -1,118 +1,121 @@
...
 class Client:
+    """Supabase client class."""
+
     def __init__(
-        self, supabaseUrl: str, supabaseKey: str, options: Optional[dict] = {}
+        self, supabase_url: str, supabase_key: str, **options,
     ):
-        """
-        Initialize a Supabase Client
+        """Instanciate the client.
```
Comment (J0) [inline supabase_py/client.py:24]: /nit instanciate -> instantiate

Not really important

**Review by J0 — APPROVED** [review]
Hey @fedden ,

Thanks for taking the time to file this PR. It's a substantial improvement over what we had previously. 
I'll file a PR back into the `supabase` branch after merging so that you won't have to keep rebasing.

Thanks much! Looking forward to getting a usable client up!

File: README.md [hunk 2]
```diff
@@ -4,35 +4,59 @@
...
-supabaseUrl=""
-supabaseKey=""
-supabase = supabase_py.Client(supabaseUrl, supabaseKey)
+
+## Usage
+```python
+import os
+from supabase_py import create_client, Client
+
+url: str = os.environ.get("SUPABASE_URL")
```
[the remark this answers is not in the document]
Comment (fedden) [reply README.md:35]: Good shout. I've added a snippet of text and I'm sure this readme will evolve as we add more functionality and can showcase the library better

File: supabase_py/client.py (hunk 1 above)
Comment (fedden) [reply supabase_py/client.py]: Good catch, fixed. Thanks

**Comment by fedden** [comment]
Thanks for the quick PR review @J0 👍 

> I'll file a PR back into the supabase branch after merging so that you won't have to keep rebasing.

Appreciate this. I wonder what the best approach would be for us to work in tandem on this client (and supporting libraries). 

If we are both iterating on the supabase upstream code, might it make sense to get write developer access to supabases upstream supabase-py, and then in an ideal world, we could just make a `develop` branch, both make feature branches off of that for whatever we are working on, and then hack, rebase, hack and merge to `develop` directly. Once we've got the client in a good place, we can merge `develop` to `master` and add it as a stable release.

What do you think?

**Comment by J0** [comment]
Hey @fedden,

> we could just make a develop branch, both make feature branches off of that for whatever we are working on, and then hack, rebase, hack and merge to develop directly. Once we've got the client in a good place, we can merge develop to master and add it as a stable release.

Having a `develop` branch as well as write access for both parties makes a lot of sense to me.  I think it sounds like a plan. I'll merge this in now and file a PR against the upstream code. Following that I'll ping the team on the Slack channel and see if they could give you write access.

**Pull request merged**